kernel/src/main/scala/cats/kernel/Order.scala

```diff
@@ -0,0 +1,210 @@
+    compare(x, y) <= 0
+
+  /**
+   * Returns true if `x` < `y`, false otherwise.
+   */
+  override def lt(x: A, y: A): Boolean =
+    compare(x, y) < 0
+
+  /**
+   * Returns true if `x` >= `y`, false otherwise.
+   */
+  override def gteqv(x: A, y: A): Boolean =
+    compare(x, y) >= 0
+
+  /**
+   * Returns true if `x` > `y`, false otherwise.
+   */
+  override def gt(x: A, y: A): Boolean =
+    compare(x, y) > 0
+
+  /**
+   * Returns a new `Order[A]` instance that first compares by the original
+   * `Order` instance and uses the provided `Order` instance to "break ties".
+   *
+   * That is, `x.whenEqual(y)` creates an `Order` that first orders by `x` and
+   * then (if two elements are equal) falls back to `y` for the comparison.
+   */
+  def whenEqual(o: Order[A]): Order[A] = new Order[A] {
+    def compare(x: A, y: A) = {
+      val c = self.compare(x, y)
+      if (c == 0) o.compare(x, y)
+      else c
+    }
+  }
+
+  /**
+   * Convert a `Order[A]` to a `scala.math.Ordering[A]`
+   * instance.
+   */
+  def toOrdering: Ordering[A] = new Ordering[A] {
+    def compare(x: A, y: A): Int = self.compare(x, y)
+  }
+}
+
+trait OrderFunctions {
+  def compare[@sp A](x: A, y: A)(implicit ev: Order[A]): Int =
+    ev.compare(x, y)
+
+  def eqv[@sp A](x: A, y: A)(implicit ev: Order[A]): Boolean =
+    ev.eqv(x, y)
+  def neqv[@sp A](x: A, y: A)(implicit ev: Order[A]): Boolean =
+    ev.neqv(x, y)
+  def gt[@sp A](x: A, y: A)(implicit ev: Order[A]): Boolean =
+    ev.gt(x, y)
+  def gteqv[@sp A](x: A, y: A)(implicit ev: Order[A]): Boolean =
+    ev.gteqv(x, y)
+  def lt[@sp A](x: A, y: A)(implicit ev: Order[A]): Boolean =
+    ev.lt(x, y)
+  def lteqv[@sp A](x: A, y: A)(implicit ev: Order[A]): Boolean =
+    ev.lteqv(x, y)
+
+  def min[@sp A](x: A, y: A)(implicit ev: Order[A]): A =
+    ev.min(x, y)
+  def max[@sp A](x: A, y: A)(implicit ev: Order[A]): A =
+    ev.max(x, y)
+}
+
+object Order extends OrderFunctions {
+
+  /**
+   * Access an implicit `Eq[A]`.
+   */
+  @inline final def apply[A](implicit ev: Order[A]) = ev
+
+  /**
+   * Convert an implicit `Order[A]` to an `Order[B]` using the given
+   * function `f`.
+   */
+  def by[@sp A, @sp B](f: A => B)(implicit ev: Order[B]): Order[A] =
+    ev.on(f)
+
+  /**
+   * Define an `Order[A]` using the given function `f`.
+   */
+  def from[@sp A](f: (A, A) => Int): Order[A] =
+    new Order[A] {
+      def compare(x: A, y: A) = f(x, y)
+    }
+
+  /**
+   * Implicitly convert a `Order[A]` to a `scala.math.Ordering[A]`
+   * instance.
+   */
+  implicit def ordering[A](implicit ev: Order[A]): Ordering[A] =
+    ev.toOrdering
+
+  /**
+   * An `Order` instance that considers all `A` instances to be equal.
+   */
+  def allEqual[A]: Order[A] =
+    new Order[A] {
+      def compare(x: A, y: A): Int = 0
+    }
+
+
+  /**
+   * A `Band[Order[A]]` can be generated for all `A` with the following
+   * properties:
+   *
+   * `empty` returns a trivial `Order[A]` which considers all `A` instances to
```

Comment:
Warning nitpicky comment: `empty` comes from `Monoid` not `Band`, so we may want to adjust the previous line of the comment to mention both `Band` and `Monoid`.